### DISCUSSION TITLE
How to get crazyflie position in C firmware easily?

### DISCUSSION THREAD

**Discussion by RavenLite** [bitcraze/discussions#623] [Q&A] [answered]
Hi!

We are doing something about slam~
So we need to get the lighthouse position and multiranger distance.
We find that there is `rangeGet` function to get the distance easily in the firmware.
So, is there similar function to get the position/coordinate?

Thanks!

**Comment by jackyyy65** [comment]
It's possible to obtain the position or coordinate of a lighthouse using a technique called "lighthouse tracking." Lighthouse tracking is a technology used in virtual reality (VR) systems that utilizes infrared sensors on a headset or controller to track the position and orientation of the device relative to one or more lighthouse base stations. The base stations emit infrared light in a pattern that the sensors on the device can detect and use to determine its position and orientation.

However, the ability to perform lighthouse tracking depends on the hardware and firmware of the device being used. If you are working with a device that supports lighthouse tracking, it may have a function or API that you can use to obtain the position or coordinate information. You may need to consult the device's documentation or firmware to determine if this is possible and how to use it.

**Reply by RavenLite** [reply]
❎ ChatGPT

**Reply by hmllr** [reply]
Indeed, this answer seems to be generated with a tool like ChatGPT - even though there might be some truth in it, those answers will not help much in solving the problem. Please do not answer in our discussions with ChatGPT, it will just generate noise and make the real and useful answers harder to find.

**Comment by hmllr** [comment, marked as answer]
Hi :)
It might be easiest to have all variables you want to access in logging variables and then read them through the logging system - an example of this is in the push demo code: https://github.com/bitcraze/crazyflie-firmware/blob/master/examples/demos/app_push_demo/src/push.c
Hope this helps (and if you need more info about the logging framework: https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/userguides/logparam/), 
Hanna

**Reply by RavenLite** [reply]
Hi,

It's really easy to get any variables defined using `logGetVarId()`.   
I have tried it and replaced `rangeGet()` with it too.

Thanks!
